packages/api/game/game.go

```diff
@@ -0,0 +1,253 @@
+package game
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/dopedao/dope-monorepo/packages/api/base"
+	"github.com/google/uuid"
+	"github.com/gorilla/websocket"
+)
+
+type Game struct {
+	Players      PlayersContainer
+	ItemEntities ItemEntitiesContainer
+
+	Register   chan *Player
+	Unregister chan *Player
+	Broadcast  chan BaseMessage
+}
+
+func (g *Game) Start(ctx context.Context) {
+	_, log := base.LogFor(ctx)
+
+	for {
+		select {
+		case player := <-g.Register:
+			g.Players.mutex.Lock()
+			g.Players.data = append(g.Players.data, player)
+
+			// handshake
+			handShakeData, err := json.Marshal(IdData{Id: player.Id.String()})
+			if err != nil {
+				log.Err(err).Msg("could not marshal handshake data")
+				return
```

Comment:
Need to make sure errors propagate back to the client. If we don't return an error code, the client wont know to cleanup/retry the websocket